### DISCUSSION TITLE
A Permanent URL of a user profile page

### DISCUSSION THREAD

**Discussion by jmatsu** [community/community#67481] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Product Feedback

### Body

Sometimes I would like to share my GitHub profile link with external media like articles, website and so on, however, this is not a permanent link and may associate with another user if I change my username on GitHub.
I know `https://github.com/<username>` looks human friendly though, could you please also provide a permanent link of a profile page? By the way, a profile icon url seems to be a permanent link unlike a profile link.

**Comment by ChidiChuks** [comment]
Your suggestion to have a permanent link to a GitHub user profile, even if the username is changed, is a valid one and could provide more stability for sharing links to GitHub profiles. While GitHub currently uses the `https://github.com/<username>` format for user profiles, they do offer a more permanent identifier for users through the user ID.

A GitHub user ID is a numeric identifier associated with each user, and it remains the same even if the username is changed. You can find your user ID by visiting your GitHub profile settings page. It's listed under "Profile" as the "GitHub ID."

For example, a user with a GitHub ID of 123456 might have a profile URL like this:

```
https://github.com/123456
```

Using the GitHub ID in the profile URL ensures that the link remains valid even if the username is changed. This can be a useful way to share a more permanent link to your GitHub profile with others.

Regarding the profile icon URL, you are correct that it typically remains the same even if other profile details are modified. You can access your profile icon URL by right-clicking on your profile picture on GitHub and selecting "Copy image address" or a similar option depending on your browser. This URL can be used as a stable reference to your profile picture.

**Reply by vinay-200508** [reply]
Please share link to your GitHub Profile (Mandatory)

**Reply by MarkCallow** [reply]
This is not working for me. I get "404 Not found". I got the id from "https://api.github.com/users/MarkCallow". It is the same as the  as the id shown in {id}+{user_name}@users.noreply.github.com in my email settings tab. How do I make it work?

**Reply by vito-torrentino** [reply]
Literally an AI-generated response. The way it described is how gitlab works.

**Comment by Darshu2934** [comment]
Darshu2934

**Comment by jfullstackdev** [comment, marked as answer]
I guess the reason why it's not implemented is because, GitHub handle is to be treated as almost permanent and not changing very often

but I've come up with a workaround but I don't know if this is a possibility for you because I doubt if you would ever want this 😆  

---

## **Step 1: Decide on a permanent link URL**

Choose a URL you control, e.g.:

```
https://yourdomain.com/github
```

This will be your “permanent GitHub profile link” that you can share anywhere. At this point, you don't want your own domain, right ? 😆 

---

## **Step 2: Fetch your current GitHub username programmatically**

GitHub numeric IDs never change. You can get the current username via the API:

```
https://api.github.com/user/47092464
```

The response JSON will contain a `"login"` field, which is your current username. For example:

```json
{
  "login": "jfullstackdev",
  "id": 47092464,
  ...
}
```

---

## **Step 3: Create a redirect page**

You can use a simple HTML + JavaScript page that fetches the username and redirects:

```html
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Redirecting to GitHub</title>
    <script>
        async function redirectToGitHub() {
            const userId = 47092464;
            const apiUrl = `https://api.github.com/user/${userId}`;
            
            try {
                const response = await fetch(apiUrl);
                if (!response.ok) throw new Error("GitHub API error");
                const data = await response.json();
                const username = data.login;
                
                // Redirect to current GitHub profile
                window.location.href = `https://github.com/${username}`;
            } catch (err) {
                console.error(err);
                document.body.innerHTML = "Unable to redirect at this time.";
            }
        }

        redirectToGitHub();
    </script>
</head>
<body>
    Redirecting to GitHub profile...
</body>
</html>
```

---

## **Step 4: Host the page**

* Upload this HTML file to your website or a serverless platform like **Netlify, Vercel, or GitHub Pages**.
* Ensure the permanent URL is easy to share, e.g., `https://yourdomain.com/github`.

---

## **Step 5: Test it**

1. Visit your permanent link.
2. The page should fetch the current username from GitHub and redirect to:

```
https://github.com/<current_username>
```

3. Change your GitHub username temporarily (if possible) and confirm the redirect still works.

---

## **Optional: Serverless / Backend Version**

If you prefer not to rely on client-side JS, you can implement a **serverless redirect function** (Node.js, Python, etc.) that fetches the username from GitHub and responds with an HTTP redirect. This ensures the redirect works even for clients without JavaScript.

enjoy coding !

**Reply by MarkCallow** [reply]
> but I've come up with a workaround but I don't know if this is a possibility for you because I doubt if you would ever want this 😆

Thank you for your thorough answer. Indeed, I have decided to use https://github.com/<current_username> as it is much the simplest and I don't expect to change my user name again.
